holonote/app/panel.py

```diff
@@ -83,41 +85,107 @@ def _create_visible_widget(self):
         if self.annotator.groupby is None:
             self.visible_widget = None
             return
-        style = self.annotator.style
-        if style.color is None and style._colormap is None:
-            data = sorted(self.annotator.df[self.annotator.groupby].unique())
-            colormap = dict(zip(data, _default_color))
-        else:
-            colormap = style._colormap
-        if isinstance(colormap, dict):
-            stylesheet = """
-            option:after {
-              content: "";
-              width: 10px;
-              height: 10px;
-              position: absolute;
-              border-radius: 50%;
-              left: calc(100% - var(--design-unit, 4) * 2px - 3px);
-              border: 1px solid black;
-              opacity: 0.5;
-            }"""
-            for i, color in enumerate(colormap.values()):
-                stylesheet += f"""
-            option:nth-child({i + 1}):after {{
-                background-color: {color};
-            }}"""
-        else:
-            stylesheet = ""
 
-        options = list(colormap)
+        style = self.annotator.style
+        self.colormap = {}
+        options = sorted(set(self.annotator.df[self.annotator.groupby].unique()))
+        # if all_options:
+        if style.color is None:
+            self.colormap = dict(zip(options, _default_color))
+        elif isinstance(style.color, str):
+            self.colormap = dict(zip(options, [style.color] * len(options)))
+        elif isinstance(style.color, hv.dim):
+            self.colormap = self.annotator.style.color.ops[0]["kwargs"]["categories"]
```

Comment:
```suggestion
            self.colormap = dict(self.annotator.style.color.ops[0]["kwargs"]["categories"])
```

Reply:
Alternatively could you do something like:

```python
colormap = dict(zip(options, style.color.apply(hv.Dataset(options, [self.annotator.groupby]))))
```

Reply:
I think @hoxbro had an initial impression that this suggestion might be too heavy for this particular use case, but I'll let him clarify/expound in his review